ADR/ADR_0001.md

```diff
@@ -0,0 +1,33 @@
+## ADR-0001: enhancement of reliability and maintainability by improving type safety 
+Date: 29-05-2025
+
+### Status: Proposed
+
+### Context
+
+There is a lot of code which contains `any` type without direct annotations or type
+narrowing, which already cause a lot of misunderstanding. 
+It makes harder and slower to implement new features and potentially 
+can cause unexpected errors.
+Besides there are also a lot of type aliases where fields are different 
+depending on some field or condition.
+Because of it code is less readable and more complex
+
+### Decision
+
+- At least use `unknown` instead of `any`
+- implement type narrowing and type guards
+- use discriminated types where possible
+- probably use "wll-known" libraries which implement common type guards
```

Comment:
The phrase "well-known libraries" feels a bit vague. Could you name one or two libraries you're considering?